hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/Http2RequestHeadersProvider.java

```diff
@@ -0,0 +1,25 @@
+package pl.allegro.tech.hermes.consumers.consumer.sender.http;
+
+import org.eclipse.jetty.http.HttpHeader;
+import pl.allegro.tech.hermes.consumers.consumer.Message;
+
+import java.util.HashMap;
+import java.util.Map;
+
+final class Http2RequestHeadersProvider implements HttpRequestHeadersProvider {
```

Comment:
In my opinion you are mixing here two different layers: HTTP and Hermes logic. Why `Http1RequestHeadersProvider` knows about Hermes specific headers such as: `Hermes-Topic-Name`, while `Http2RequestHeadersProvider` assumes that `baseRequestHeadersProvider` provides all necessary headers and only guards compatibility with HTTP/2 protocol specification? It seems that those two implementations are too tightly coupled to exist separate.

What do you think about having one implementation of headers provider parametrized by forbidden headers?

Reply:
I've commited some changes, check it out :)